### PR TITLE
Update opencensus library

### PR DESCRIPTION
**Pull request by bianpengyuan**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -611,13 +611,13 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        version = "7877337633466358ed680f9b26967da5b310d7aa",
-        sha256 = "12ff300fa804f97bd07e2ff071d969e09d5f3d7bbffeac438c725fa52a51a212",
+        version = "ba631066779a534267fdb1321b19850eb2b0c000",
+        sha256 = "f239a40803f6e2e42b57c9e68771b0990c4ca8b2d76b440073cdf14f4211ad26",
         strip_prefix = "opencensus-cpp-{version}",
         urls = ["https://github.com/census-instrumentation/opencensus-cpp/archive/{version}.tar.gz"],
         use_category = ["observability_ext"],
         extensions = ["envoy.tracers.opencensus"],
-        last_updated = "2020-06-01",
+        last_updated = "2020-10-07",
```
Comment (htuch) [inline bazel/repository_locations.bzl:620]: We're aiming to make this the date that repository_locations.bzl was last updated, so could you bump that? Thanks.

Comment (bianpengyuan) [reply bazel/repository_locations.bzl]: Done.

Comment (moderation) [reply bazel/repository_locations.bzl]: No! I believe that `last_updated` should reflect the date of the commit or release (and we need to update the policy to state this). We need to get our story straight before EnvoyCon @htuch :smile:

Comment (htuch) [reply bazel/repository_locations.bzl]: @moderation and I chatted about this offline and we've agreed on a less ambiguous `release_date` to replace `last_update`. I'll update these in bulk, so no need to worry about it in this PR or those merging today.

**Pull request merged**